index.src.html

```diff
@@ -3182,6 +3186,8 @@
   readonly attribute long long? timestamp;          // microseconds
   readonly attribute VideoColorSpace colorSpace;
 
+  VideoFrameMetadata getMetadata();
```

Comment:
Should this perhaps be an attribute? A method may be safer since we can always return a new copy and then mutations won't be a risk, but an attribute seems more natural to me.

Reply:
I assumed each frame would create an immutable copy, so this could be a SameObject readonly field. If that's true then an attribute sounds preferable to me for consistency of the rest of the fields.

Reply:
VideoFrameMetadata is a dictionary right now.
I guess we could make it an interface to put it as an attribute.

Reply:
Are both dictionaries and interfaces things that can have `partial` implementations?

Reply:
Yes, they can both be `partial`.

Reply:
I don't think it's going to be possible to deeply freeze deserialized data, because for example `ArrayBuffer` is always mutable. Given that, I have become supportive of staying with a method and deserializing a new copy each time it is called.

Reply:
Agreed, let's mark this conversation as resolved.